packages/dds/tree/src/test/feature-libraries/contextuallyTyped.spec.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { strict as assert } from "assert";
+
+import { ValueSchema } from "../../core";
+import { isPrimitiveValue } from "../../feature-libraries";
+// Allow importing from this specific file which is being tested:
+/* eslint-disable-next-line import/no-internal-modules */
+import { allowsValue } from "../../feature-libraries/contextuallyTyped";
+
+describe("ContextuallyTyped", () => {
+	it("isPrimitiveValue", () => {
+		assert(isPrimitiveValue(0));
+		assert(isPrimitiveValue(0.001));
+		assert(isPrimitiveValue(NaN));
+		assert(isPrimitiveValue(true));
+		assert(isPrimitiveValue(false));
+		assert(isPrimitiveValue(""));
+		assert(!isPrimitiveValue({}));
+		assert(!isPrimitiveValue(undefined));
+		assert(!isPrimitiveValue(null));
+		assert(!isPrimitiveValue([]));
+	});
+
+	it("allowsValue", () => {
```

Comment:
are `null` or `[]` relevant here? Should they have test cases?

Reply:
the implementation just uses typeof and both those are objects, but testing them is not really gaining anything. I just left them in the above code since it was being moved